content/docs/2_cookbook/9_setup/0_caddy/cookbook-recipe.txt

```diff
@@ -0,0 +1,124 @@
+	}
+	redir @blocked /
+}
+
+:80 { # Adjust to your setup
+	import common
+	import kirby
+	root * /usr/share/caddy # Adjust to your setup
+}
+
+```
+
+## Line-by-line explanation
+
+```
+(common) {
+	...
+}
+
+(kirby) {
+	...
+}
+```
+We're creating two custom **snippets** here with the names `common` and `kirby`. Snippets are blocks of configuration directives, which we can later reference and reuse. It's good style to put application-specific settings into their own snippets to know why given settings were initially put in place. So we're splitting our settings into a `common` snippet and `kirby` snippet.
+<br/>
+
+```
+php_fastcgi php:9000
+```
+
+This is the line that does the [handover to the PHP interpreter](https://caddyserver.com/docs/caddyfile/directives/php_fastcgi). We're using the modern FastCGI interface for this handover. The `php_fastcgi` directive takes a network address or unix socket as a parameter, where a PHP-FPM process is listening and waiting for scripts to be run. So this setting depends on your specific setup. If you're using docker, you can just put down the name and port of your FPM container (e.g. `php:9000`). If you're running PHP-FPM on the same system, you can use `localhost` followed by the port number.
+This directive also comes with a lot of good default values for PHP-specific configuration values (e.g. it automatically sets `try_files` to contain an `index.php` file – so a request gets forwarded to PHP when there is no file with a specific name)
+<br/>
+
+```
+tls name@user.com
+```
+
+This directive will take care of your TLS certificate. If you want to use a free certificate from Let's Encrypt, just enter your E-Mail address here and you're done. If you want to run your Kirby page locally, just use `tls internal` for a self-signed certificate.
+<br/>
+
+
+```
+file_server
+```
+
+This directive will make sure Caddy serves all static assets, e.g. all of the JavaScript, font files, icons, images, etc. Without this directive, only PHP files would be served/interpreted.
+<br/>
+
+
+```
+@blocked {
+	path *.txt *.md /content/* /site/* /kirby/* /.*
+}
+redir @blocked /
+```
+
+Here we are defining a custom matcher – which is a statement that we can then use in multiple locations as an argument for other directives. Our matcher is called `@blocked` – because we're defining paths which should not be accessible by website visitors: all raw txt/md files and everything in the `content`, `site` or `kirby` folders, as well as hidden files. In the following line, we're redirecting all these requests to the home page of our site.
+<br/>
+
+
+```
+:80 {
+	...
+}
+```
+This is a site block, which is the top level component of a Caddyfile. The block begins with the address of our server. We're just defining a port number (80) here, which means that our server will listen on all its interfaces. Instead you can also put an IP address or a domain name here, depending on your server setup.
```

Comment:
Using `:80` will disable the HTTPS redirect IIRC, so in my understanding it conflicts with the `tls <email>` directive. Maybe I've misremembered it.

In a container with a reverse-proxy attached to it, `:80` is likely what we want, but then without TLS handling by Caddy. In other scenarios I've only seen the domain name used. Using a domain name would also allow to show that multiple sites with different domains and roots can be served from the same Caddy instance.

Reply:
You're right. I've changed it.